uxarray/grid/geometry.py

```diff
@@ -1561,3 +1568,174 @@ def inverse_stereographic_projection(x, y, central_lon, central_lat):
+            Cartesian coordinate of the point
+        inclusive : bool
+            Flag to determine whether to include points on the nodes and edges of the face
+
+    Returns
+    -------
+    bool
+        True if point is inside face, False otherwise
+    """
+
+    # Validate the inputs
+    if len(edges_xyz[0][0]) != 3:
+        raise ValueError("`edges_xyz` vertices must be in Cartesian coordinates.")
+
+    if len(point_xyz) != 3:
+        raise ValueError("`point_xyz` must be a single [3] Cartesian coordinate.")
+
+    # Initialize the intersection count
+    intersection_count = 0
+
+    # Set to hold unique intersections
+    unique_intersections = set()
+
+    location = _classify_polygon_location(edges_xyz)
+
+    if location == 1:
+        ref_point_xyz = np.array([0.01745241, 0.0, -0.9998477], dtype=np.float64)
+    elif location == -1:
+        ref_point_xyz = np.array([0.01745241, 0.0, 0.9998477], dtype=np.float64)
+    else:
+        ref_point_xyz = np.array([0.01745241, 0.0, -0.9998477], dtype=np.float64)
+
+    # Initialize the points arc between the point and the reference point
+    gca_cart = np.empty((2, 3), dtype=np.float64)
+    gca_cart[0] = point_xyz
+    gca_cart[1] = ref_point_xyz
+
+    # Loop through the face's edges, checking each one for intersection
+    for ind in range(len(edges_xyz)):
+        # If the point lies on an edge, return True if inclusive
+        if point_within_gca(
+            point_xyz,
+            edges_xyz[ind][0],
+            edges_xyz[ind][1],
+        ):
+            if inclusive:
+                return True
+            else:
+                return False
+
+        # Get the number of intersections between the edge and the point arc
+        intersections = gca_gca_intersection(edges_xyz[ind], gca_cart)
+
+        # Add any unique intersections to the intersection_count
+        for intersection in intersections:
+            intersection_tuple = (
+                intersection[0],
+                intersection[1],
+                intersection[2],
+            )
+            if intersection_tuple not in unique_intersections:
+                unique_intersections.add(intersection_tuple)
+                intersection_count += 1
+
+    # Return True if the number of intersections is odd, False otherwise
+    return intersection_count % 2 == 1
+
+
+@njit(cache=True)
+def _find_faces(face_edge_cartesian, point_xyz, inverse_indices):
+    """Finds the faces that contain a given point, inside a subset "face_edge_cartesian"""
+
+    index = []
+
+    # Loop through the whole subset
+    for i, face in enumerate(face_edge_cartesian):
+        # Check if the point is inside the face
+        contains_point = point_in_face(
+            face,
+            point_xyz,
+            inclusive=True,
+        )
+
+        # If the point is, add it to the list of faces
+        if contains_point:
+            index.append(inverse_indices[i])
+
+    return index
+
+
+def get_max_face_radius(self):
```

Comment:
This should be `_populate_max_face_radius` similar to our other population methods.